lib/_sendEvent.ts

```diff
@@ -20,6 +20,10 @@ export function makeSendEvents(requestFn: RequestFnType) {
       );
     }
 
+    if (!this._userToken && this._anonymousUserToken) {
+      this.setAnonymousUserToken(true);
+    }
```

Comment:
interesting time to do this check, but it does make sense that you want to do it as late as possible.